src/otx/core/model/entity/anomaly.py

```diff
@@ -0,0 +1,283 @@
+"""Base Anomaly OTX model."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from abc import ABC, abstractproperty
+from dataclasses import dataclass
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+import onnx
+import openvino
+import torch
+from anomalib import TaskType
+from torch import nn
+
+from otx.core.data.dataset.base import LabelInfo
+from otx.core.data.entity.anomaly import (
+    AnomalyClassificationBatchPrediction,
+    AnomalyDetectionBatchPrediction,
+    AnomalySegmentationBatchPrediction,
+)
+from otx.core.data.entity.base import T_OTXBatchDataEntity
+from otx.core.exporter.base import OTXModelExporter
+from otx.core.model.entity.base import OTXModel, OVModel
+from otx.core.types.precision import OTXPrecisionType
+
+if TYPE_CHECKING:
+    from openvino.model_api.models import Model
+    from openvino.model_api.models.anomaly import AnomalyResult
+    from torchvision.transforms.v2 import Transform
+
+
+class _AnomalibLightningArgsCache:
```

Comment:
The problem is that in Anomalib, the parameters are passed to the lightning model. But since we create a torch (OTX) model and pass it to the engine/config.yaml, we need a way to pass parameters to the lightning model. The idea behind this is to cache all the arguments that are passed to the OTXModel and read them in the lightning model. I don't know a better workaround for this.

Reply:
Then how about reuse this class (`src/otx/core/utils/cache.py::TrainerArgumentsCache`) to eliminate duplication?